internal/lsp/server.go

```diff
@@ -377,6 +384,31 @@ func (s *Server) handleDefinition(req *lsproto.RequestMessage) error {
 	return s.sendResult(req.ID, &lsproto.Definition{Locations: &lspLocations})
 }
 
+func (s *Server) handleCompletion(req *lsproto.RequestMessage) (messageErr error) {
+	params := req.Params.(*lsproto.CompletionParams)
+	file, project := s.getFileAndProject(params.TextDocument.Uri)
+	pos, err := s.converters.LineAndCharacterToPositionForFile(params.Position, file.FileName())
+	if err != nil {
+		return s.sendError(req.ID, err)
+	}
+
+	// !!! remove this after completions is fully ported/tested
+	defer func() {
+		if r := recover(); r != nil {
+			stack := debug.Stack()
+			messageErr = s.sendError(req.ID, fmt.Errorf("panic obtaining completions: %v\n%s", r, string(stack)))
```

Comment:
I don't know if this is the best way to do this, but it's the way I could find online.

Reply:
I would print the error to the output and just return an empty completion list. Sending an explicit error means a stack of errors in the editor that pop up when typing.